### ISSUE TITLE
Self-assigning function declaration containing `eval()` serialized incorrectly

### ISSUE THREAD

**Issue by overlookmotel** [overlookmotel/livepack#328]
Input:

```js
function x() {
  eval('0');
  x = 1;
}
export default x;
```

Output:

```js
export default (0, eval)(`
  'use strict';
  x => function x() { eval('0'); x = 1; };
`)();
```

Calling original function sets `x`. Calling function as serialized by Livepack throws `TypeError: Assignment to constant variable`.

Problem is that in output, external var `x` has its name frozen (due to `eval()`) and is shadowed by the function name. Function is a function expression so assignment to its name is illegal.

Correct output would be:

```js
export default (0, eval)(`
  'use strict';
  x => x = function() { eval('0'); x = 1; };
`)();
```

Here, function expression is anonymous, and has its `name` property set implicitly by the preceding `x =` assignment.

**Comment by overlookmotel** [comment]
Tests (currently disabled as are failing) are at:

https://github.com/overlookmotel/livepack/blob/2e4f21a642b01e7d803e1f57ddab8f8a342b4564/test/functions.test.js#L6200

**Comment by overlookmotel** [comment]
This is being fixed as part of implementation of #65 - work in progress on [babel-record-asts](https://github.com/overlookmotel/livepack/tree/babel-record-asts) branch. These tests are now passing on that branch.
